src/tools/compiletest/src/runtest.rs

```diff
@@ -992,7 +997,7 @@ actual:\n\
     fn check_expected_errors(&self,
                              expected_errors: Vec<errors::Error>,
                              proc_res: &ProcRes) {
-        if proc_res.status.success() {
+        if proc_res.status.success() && expected_errors.iter().any(|x| x.kind == Some(ErrorKind::Error)) {
```

Comment:
I think we may want to thread through some more information here, this means a compile-fail test which only checked for warnings would pass if compilation succeeded, I believe.

Perhaps whether or not `proc_res.status.success()` should be checked could be passed in or lifted out?